recipes/xerces-c/all/conanfile.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+
+from conans import ConanFile, CMake, tools
+import os
+
+
+class XercesCConan(ConanFile):
+    name = "xerces-c"
+    description = "Xerces-C++ is a validating XML parser written in a portable subset of C++"
+    topics = ("conan", "xerces", "XML", "validation", "DOM", "SAX", "SAX2")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "http://xerces.apache.org/xerces-c/index.html"
+    license = "Apache-2.0"
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        # https://xerces.apache.org/xerces-c/build-3.html
+        # TODO : check if we need options for that variants
+        cmake.definitions["network-accessor"] = {"Windows": "winsock",
```

Comment:
I guess it's better to throw an errors here for unsupported OS

Reply:
Which OS did you mean as unsupported? I thought all three platforms worked well with those options for built-in OS engines. For this reason, I had disabled auto-finding external engines (curl, ICU).

Reply:
e.g. if someone tries to build for `Android`, he will receive an obscure error
better to print a friendly message earlier

Reply:
Whitelisted those three platforms in `configure`.